main/lsp/LSPTypechecker.cc

```diff
@@ -240,10 +324,28 @@ bool LSPTypechecker::runSlowPath(LSPFileUpdates updates, WorkerPool &workers, bo
             ENFORCE(tree.file.exists());
             affectedFiles.push_back(tree.file);
         }
-        if (finalGS->sleepInSlowPath) {
+        if (gs->sleepInSlowPath) {
             Timer::timedSleep(3000ms, *logger, "slow_path.typecheck.sleep");
         }
 
+        // Inform the fast path that this global state is OK for typechecking as resolution has completed.
```

Comment:
This satisfies an ENFORCE we have in various places, which checks via this property that resolution has completed.